hack/e2e-common.sh

```diff
@@ -0,0 +1,181 @@
+max_tries=60
```

Comment:
Note to reviewers: There's quite a bit of cleanup/refactoring that could be done here, but I explicitly didn't do that so it'd be easier to review: use your favorite visual diff tool to compare this file to the original e2e-test.sh.

Reply:
do you mind moving this move to separate commit. that would be a lot more read able.

Reply:
I'm not sure how much more readable it would make anything. That commit would simply comprise e2e-common.sh and e2e-test.sh in the exact form they exist in this PR.

Reply:
Possible misunderstanding here, refactoring that *could* be done, but isn't in this PR.

Reply:
Correct: I kept the existing code from e2e-test.sh, warts and all, and just moved it to e2e-common.sh with as little change as possible, precisely to make this PR more scrutable. We can refactor/reorganize later.